odysseus/odysseus_tree/board/tpu/plugin_user_property_timestamp.c

```diff
@@ -0,0 +1,48 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/time.h>
+
+
+#include "nanomq/plugin.h"
+
+/*
+ * How to compile:
+ * gcc -I../ -fPIC -shared plugin_user_property.c -o plugin_user_property.so
+ */
+
+
+int cb(void *data)
+{
+	char **property = data;
+	if (property != NULL) {
+		
+		// key of ts, add 1 bc termination byte
+		property[0] = malloc(strlen("ts") + 1);
+		strcpy(property[0], "ts");
+		
+		struct timeval tv;
+  
+		gettimeofday(&tv, NULL);
```

Comment:
is there not a function to just get the date in Unix time in a numerical format?

Reply:
Doing the "time since epoch" calc manually feels a bit funky, is it because we want millisecond precision? If so we should call that out in the plugin

Reply:
not that I could find.